### PR TITLE
blockdev_inc_backup_nospace_with_bitmap_mode: add "yes" when mkfs disk

### PR DESCRIPTION
**Pull request by aliang123**
Add "yes" to mkfs when format a disk

Signed-off-by: Aihua Liang <aliang@redhat.com>
id:2061234

### CODE REVIEW

**Comment by aliang123** [comment]
(1/3) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.q35: STARTED
| (1/3) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.q35: PASS (767.54 s)
 (2/3) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.blockdev_inc_backup_nospace_with_bitmap_mode.on_success.q35: STARTED
| (2/3) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.blockdev_inc_backup_nospace_with_bitmap_mode.on_success.q35: PASS (115.23 s)
 (3/3) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.blockdev_inc_backup_nospace_with_bitmap_mode.always.q35: STARTED
/ (3/3) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.blockdev_inc_backup_nospace_with_bitmap_mode.always.q35: PASS (84.84 s)

**Comment by aliang123** [comment]
@zhencliu @MiriamDeng please help to review, thanks.

File: qemu/tests/cfg/blockdev_inc_backup_nospace_with_bitmap_mode.cfg [hunk 1]
```diff
@@ -42,7 +42,7 @@
 
     tmp_image_file = /tmp/tmp_image_file
     pre_command = "mkdir -p ${inc_path} && dd if=/dev/urandom of=${tmp_image_file} bs=1M count=100 &&"
-    pre_command += " mkfs.ext4 ${tmp_image_file} && mount -o loop ${tmp_image_file} ${inc_path}"
+    pre_command += " yes|mkfs.ext4 ${tmp_image_file} && mount -o loop ${tmp_image_file} ${inc_path}"
```
Comment (zhencliu) [inline qemu/tests/cfg/blockdev_inc_backup_nospace_with_bitmap_mode.cfg:45]: We use yes when there is already a fs created, e.g. 

# mkfs.ext4 /tmp/lzc/tmp_image_file 
mke2fs 1.45.6 (20-Mar-2020)
/tmp/lzc/tmp_image_file contains a ext4 file system
	last mounted on Thu May 19 03:39:08 2022
Proceed anyway? (y,N) ^C

But as /tmp/lzc/tmp_image_file  is newly created, yes is not required, e.g.

# dd if=/dev/urandom of=/tmp/lzc/tmp_image_file bs=1M count=100
100+0 records in
100+0 records out
104857600 bytes (105 MB, 100 MiB) copied, 1.87364 s, 56.0 MB/s
# mkfs.ext4 /tmp/lzc/tmp_image_file
mke2fs 1.45.6 (20-Mar-2020)
Discarding device blocks: done                            
Creating filesystem with 102400 1k blocks and 25688 inodes
Filesystem UUID: e3d1123b-33e0-49fa-a393-7be56b0729c9
Superblock backups stored on blocks: 
	8193, 24577, 40961, 57345, 73729

Allocating group tables: done                            
Writing inode tables: done                            
Creating journal (4096 blocks): done
Writing superblocks and filesystem accounting information: done 

Please refer to more comments in bz, thanks.

**Comment by MiriamDeng** [comment]
Will post the test result soon.

**Comment by MiriamDeng** [comment]
The test log is as following
 (1/2) Host_RHEL.m8.u7.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.7.0.ppc64le.io-github-autotest-qemu.blockdev_inc_backup_nospace_with_bitmap_mode.on_success: STARTED
/ (1/2) Host_RHEL.m8.u7.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.7.0.ppc64le.io-github-autotest-qemu.blockdev_inc_backup_nospace_with_bitmap_mode.on_success: FAIL: Count of bitmap changed after inc-backup (167.14 s)
 (2/2) Host_RHEL.m8.u7.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.7.0.ppc64le.io-github-autotest-qemu.blockdev_inc_backup_nospace_with_bitmap_mode.always: STARTED
\ (2/2) Host_RHEL.m8.u7.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.7.0.ppc64le.io-github-autotest-qemu.blockdev_inc_backup_nospace_with_bitmap_mode.always: PASS (138.54 s)

**Comment by aliang123** [comment]
(1/2) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.blockdev_inc_backup_nospace_with_bitmap_mode.on_success.q35: STARTED
| (1/2) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.blockdev_inc_backup_nospace_with_bitmap_mode.on_success.q35: PASS (86.09 s)
 (2/2) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.blockdev_inc_backup_nospace_with_bitmap_mode.always.q35: STARTED
| (2/2) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.blockdev_inc_backup_nospace_with_bitmap_mode.always.q35: PASS (83.64 s)

**Comment by aliang123** [comment]
@zhencliu can you help to review again, thanks.

**Comment by aliang123** [comment]
Having discussed with zhenchao, the error is just as expected. Test env issue should be resolved by manual, not by test case.
So close the patch and will close the auto bug later.

**Pull request closed without merging**